src/crab_verifier.cpp

```diff
@@ -95,8 +95,14 @@ static checks_db generate_report(cfg_t& cfg, const crab::invariant_table_t& pre_
     }
 
     if (thread_local_options.check_termination) {
-        const auto last_inv = post_invariants.at(cfg.exit_label());
-        m_db.max_loop_count = last_inv.get_loop_count_upper_bound();
+        // Calculate the maximum loop count across all post-invariants to detect potential
+        // infinite loops or excessive iterations in any part of the program.
```

Comment:
Is this true? If I'm not mistaken, `m_db.max_loop_count` is now only there for the report.